### DISCUSSION TITLE
Noise Reduction

### DISCUSSION THREAD

**Discussion by hypeARM** [rtl-airband/RTLSDR-Airband#352] [Ideas]
Hi,

If rtl_airband is able to calculate noise level, why doesn't it already have a noise reduction module? 🤔

**Comment by charlie-foxtrot** [comment]
The noise level that `rtl_airband` calculates the RF noise floor.  This is the amount of energy that is being received at each of the configured frequencies when there is no transmission.  The amount of energy being received when there is a transmission is the signal level.  These are both measurements of RF energy, not audio quality.

When there is a detected transmission, the received energy is passed through a demodulator to generate audio.  There is no detection, measurement, or attempts at reducing audio noise after demodulation.  If this is something you're interested in doing I would recommend looking at audio processing software that could consume the `.mp3` files that are being generated.

**Reply by hypeARM** [reply]
I don't know of an easy, simple way to do this just by taking the output MP3 file and going back to Icecast

**Comment by feelingwalnut** [comment]
what is your noise? 
static? electrical? competing frequencies?

**Reply by hypeARM** [reply]
Static, noise floor & FM Broadcasting

Remember my RTLSDR is able to filter and only pass 118-136MHz

**Reply by feelingwalnut** [reply]
have you tried grounding and a better power supply, an fm block filter or a lna for weak signal?
or a tuned antenna?

**Reply by hypeARM** [reply]
I already have a good quality grounding for the RF, on power supply I am using is from the raspberry pi, I have already tried with other power supply and the noise remained, my antenna is tuned to 118-136MHz and its center is at 127MHz

The issue may be the following, I am approximately 12km away from TWR

The antenna I use is AirNav's VHF Airband and the RTL-SDR is AirNav's FlightStick

My noise floor is around -52dBFS while the strongest signal hovers around -20dBFS

**Reply by feelingwalnut** [reply]
hmm, hopefully someone who knows more will be helpful but best i can recommend with that is more height, get the antenna up as much as possible.
try tweaking your gain or center frequency to see if there are any sweet spots. consider a LNA and maybe move to a yagi.
if you are getting noise from fm radio, consider a filter.

in my neck of the woods the 100mhz range is used for some emergency stuff and it sounds kind of garbage but gets much deeper into the forest than the 400mhz range. even in a good signal area, it just doesnt sound as clear. don't know how to go about on-the-fly "enhancing" the audio output because i don't do streaming.

**Reply by hypeARM** [reply]
Thnx for ideas ;)

No problem, I must migrate my station to the Airport side so I will get TWR and GND, with high clarity and quality!
